sdk/digitaltwins/azure-digitaltwins-core/API design.md

```diff
@@ -525,6 +524,358 @@ These APIs are invoked via DigitalTwinsClient.
 ```
 </details>
 
+## Digital Twins
+<details><summary><b>Terminology</b></summary>
+
+A digital twin is an instance of one of your custom-defined models.
+
+</details>
+
+<details><summary><b>Async APIs</b></summary>
+
+These APIs are invoked via DigitalTwinsAsyncClient.
+
+```java
+/**
+ * Gets a digital twin.
+ *
+ * @param digitalTwinId The Id of the digital twin. The Id is unique within the service and case sensitive.
+ * @return The application/json digital twin
+ */
+@ServiceMethod(returns = ReturnType.SINGLE)
+public Mono<string> getDigitalTwin(String digitalTwinId)
+
+/**
+ * Gets a digital twin.
+ *
+ * @param digitalTwinId The Id of the digital twin. The Id is unique within the service and case sensitive.
+ * @param modelClass The model class to convert the response to.
+ * @return A Http response containing the application/json digital twin
+ */
+@ServiceMethod(returns = ReturnType.SINGLE)
+public <T> Mono<T> getDigitalTwinWithResponse(String digitalTwinId, Class<T> modelClass)
```

Comment:
Shouldn't this be `<T> Mono<Response<T>>` ?

Reply:
+1
On that note, fyi - we are going to be adding a custom response type (inherited from `ResponseBase<H, T>`), which we will return from these public APIs (for APIs that should return json string). Strongly typed objects can be returned as-is from the PL.

Reply:
Good point, let me sync with @abhipsaMisra on this.